content/200-concepts/100-components/01-prisma-schema/06-relations/420-referential-integrity.mdx

```diff
@@ -8,21 +8,53 @@ preview: true
 
 <TopBlock>
 
-Referential integrity means a set of constraints or triggers to handle the consistency of data of a [`@relation`](/reference/api-reference/prisma-schema-reference#relation). It can mean preventing a deletion of a record referred from other records, or it can be a cascading action defined in the field's [`referential actions`](/concepts/components/prisma-schema/relations/referential-actions). 
+To keep data consistent, it is necessary to have rules for what happens when related records are deleted or updated. For example, a `Post` model will need to define what happens if its author is deleted. **Referential integrity** refers to the set of constraints and triggers used to enforce this consistency.
 
-The referential integrity can either be handled in the database using foreign keys, or in the Prisma Client. There are pros and cons on both. In general, if the database supports foreign keys, it is usually the preferred choice and the default for SQL databases when using Prisma.
+In Prisma, relations between records are defined using the [`@relation`](/reference/api-reference/prisma-schema-reference#relation) attribute. The `onUpdate` and `onDelete` arguments of this attribute can be used to define which **referential actions** take place on updating or deleting the related record.
 
-In certain cases, either when the database doesn't support foreign keys, or when they are not preferred due to scaling reasons, Prisma can disable them and handle the referential integrity in the client.
+For example, take the following blog post schema, with a one-to-many relation between `User`s and `Post`s:
+
+```prisma file=schema.prisma highlight=4;normal
+model Post {
+  id       Int    @id @default(autoincrement())
+  title    String
+  author   User   @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+In this case, the `onDelete: Cascade` argument specifies the `Cascade` action, deleting the post if the related user is deleted. See [Referential actions](/concepts/components/prisma-schema/relations/referential-actions) for further details of supported actions.
+
+When connecting to relational databases that support foreign keys, referential integrity can be handled in the underlying database. This is the default option in Prisma when using SQL databases.
+
+Alternatively, Prisma can emulate referential integrity in the client. This is the default option for MongoDB, which does not support foreign keys, and can also be selected for relational databases in cases where foreign keys are not supported, or not preferred due to scaling reasons.
+
+<Admonition type="warning">
+  Be aware that currently Prisma only implements the referential actions in this
+  mode, but not the constraints. If you set the referential actions to `NoAction`
+  there will be no checks to prevent you from breaking the referential
+  integrity.
```

Comment:
Does this really belong in this document or in the "ref actions" one where we describe the emulated ref actions?

Reply:
I've moved this to the `SetAction` section of the referential actions page and rewritten it a bit - let me know if the new version makes sense, and if it would be better here or somewhere else on the page.